flake.nix

```
@@ -95,7 +103,10 @@
         inherit lanzatool;
       };
 
-      nixosModules.lanzaboote = import ./nix/lanzaboote.nix;
+      nixosModules.lanzaboote = { pkgs, lib, ... }: {
+        imports = [ ./nix/lanzaboote.nix ];
+        boot.lanzaboote.package = lib.mkDefault self.packages.${pkgs.system}.lanzaboote;
```

The naming is a bit awkward, but this needs to be `lanzatool` to avoid breaking the tests:

```suggestion
        boot.lanzaboote.package = lib.mkDefault self.packages.${pkgs.system}.lanzatool;
```

Are you overriding this here so people using the module don't have to use the overlay?

---

Yes. Overlays are pretty bad from a performance perspective (at least 500MB RAM for evaluation). Also they are harder to understand and not needed here. In most cases they actually shouldn't be used (i.e. when adding new attributes to nixpkgs)